datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -247,6 +247,14 @@ def create_scraper_configuration(self, instance=None):
         config['type_overrides'] = default_instance.get('type_overrides', {})
         config['type_overrides'].update(instance.get('type_overrides', {}))
 
+        # `_type_override_patterns` is a dictionary where we store Pattern objects
+        # that match metric names as keys, and their corresponding metric type overrrides as values.
+        config['_type_override_patterns'] = {}
+
+        for metric, type in iteritems(config['type_overrides']):
+            if '*' in metric:
+                config['_type_override_patterns'][compile(translate(metric))] = type
```

Comment:
From a complexity and performance perspective, I think we can do even better.

Here, we have a map with one entry per “pattern” `type_override` even if several entries have the same type.
It means that, in the loop below, we’ll do several regex matching where a single regex match could be enough.

The idea would be for `config['_type_override_patterns']` to have a single entry per type.

That would be something like this:
```suggestion
        overrides_per_type = {}
        for metric, type in iteritems(config['type_overrides']):
            if '*' in metric:
                if type not in overrides_per_type:
                    overrides_per_type[type] = []
                overrides_per_type[type].append(fnmatch.translate(metric))
                
        for type, metric_patterns in iteritems(overrides_per_type):
            config['_type_override_patterns'][compile('|'.join(metrics_patterns))] = type
```

Reply:
thanks @L3n41c for the suggestion!
I actually thought about this approach at first but I'm not sure if it's worth it, as this path is executed only once (at the class creation) and more importantly this makes the code less straight forward and more difficult to read/understand.
@ofek what do you think about the tradeoff perf/code-readability here? is it worth it?

Reply:
If only at class creation time, then yes might not be worth it

Reply:
thank you, mind approving the PR please if looks good then?

Reply:
@ofek a friendly bump!